### PR TITLE
Devtooling 249: Fixing user_roles and group_roles resource tests

### PR DESCRIPTION
**Pull request by charliecon**

### CODE REVIEW

File: genesyscloud/user_group_role_common.go [hunk 1]
```diff
@@ -73,88 +77,123 @@ func readSubjectRoles(subjectID string, authAPI *platformclientv2.AuthorizationA
...
 		role := make(map[string]interface{})
 		role["role_id"] = roleID
-		role["division_ids"] = divs
+		role["division_ids"] = addDivisionIdsSetToRole(d, divs, roleID, homeDivId)
 		roleSet.Add(role)
 	}
 	return roleSet, resp, nil
 }
 
-func updateSubjectRoles(ctx context.Context, d *schema.ResourceData, authAPI *platformclientv2.AuthorizationApi, subjectType string) diag.Diagnostics {
```
Comment (charliecon) [inline genesyscloud/user_group_role_common.go:82]: The only change to this function was taking everything outside of the if statement `if d.HashChange("roles") {..}` and instead returning at the beginning if `!d.HasChange("roles")` or if `rolesConfig == nil`

File: genesyscloud/user_group_role_common.go [hunk 2]
```diff
@@ -73,88 +77,123 @@ func readSubjectRoles(subjectID string, authAPI *platformclientv2.AuthorizationA
...
+				return resp, diag.Errorf("Failed to add role grants for subject %s: %s", d.Id(), err)
 			}
+			return nil, nil
+		})
+		if diagErr != nil {
+			return diagErr
 		}
 	}
 	return nil
 }
```
Comment (charliecon) [inline genesyscloud/user_group_role_common.go:164]: The most important change is described in these comments

File: genesyscloud/user_group_role_common.go [hunk 3]
```diff
@@ -73,88 +77,123 @@ func readSubjectRoles(subjectID string, authAPI *platformclientv2.AuthorizationA
...
 
+// If the user provides no division ids, we add the home division to that set for them. Previously, we had division_ids: Computed
+// to avoid errors. This only caused more problems with testing because division_ids would always cause a plan not empty error,
+// going from ["<home division ID>"] to [(known after apply)]
+// Solution: Remove the computed attribute from schema and use the function below to set the division_ids field on read.
+// addDivisionIdsSetToRole - checks if the home division was already included in the division_ids set in the local config resource schema
+// If yes, set it as such on the read. If not, do not set it back in on the read.
+func addDivisionIdsSetToRole(d *schema.ResourceData, divIdsFromApi *schema.Set, roleId, homeDivId string) *schema.Set {
```
Comment (charliecon) [inline genesyscloud/user_group_role_common.go:171]: The most important change is described in these comments

**Review by carnellj-genesys — APPROVED** [review]
